### DISCUSSION TITLE
Implicit flow password reset

### DISCUSSION THREAD

**Discussion by bflemi3** [supabase/supabase#27998] [Questions]
I'm following the instructions found on this [page](https://supabase.com/docs/guides/auth/passwords#resetting-a-password) from your docs. My app is a React 18 SPA using react-router-dom v6.

After pressing the link in the email and being taken to the password reset page I created, I create a new password, then try to update the password:

```typescript
const { data, error } = await supabase.auth.updateUser({ password })
...
```

However, the `user` in `data` is not set and the `error` hydrated with:

> 'AuthSessionMissingError: Auth session missing!\n    
at http://dev-poa.localhost:3000/node_modules/.vite/deps/@supabase_supabase-js.js?v=6a34fa3d:5420:17\n   
at SupabaseAuthClient._useSession (http://dev-poa.localhost:3000/node_modules/.vite/deps/@supabase_supabase-js.js?v=6a34fa3d:5297:20)\n    
at async SupabaseAuthClient._updateUser (http://dev-poa.localhost:3000/node_modules/.vite/deps/@supabase_supabase-js.js?v=6a34fa3d:5414:14)\n    
at async http://dev-poa.localhost:3000/node_modules/.vite/deps/@supabase_supabase-js.js?v=6a34fa3d:5409:14'

![Screenshot 2024-07-15 at 1 23 49 PM](https://github.com/user-attachments/assets/698bc97b-c336-4e56-be4c-2a5521b11c71)

I'm reaching this page as an unauthenticated user that needs to reset their password because they've forgotten it. I should also mention that I've added the correct RedirectURLs so I don't think that's the issue.

How do I properly allow a user to reset their password if they're not authenticated?

**Comment by silentworks** [comment]
You should share the code you used for your `resetPasswordByEmail` here. I have a demo project here doing what the page in the docs state and it works as explained https://github.com/silentworks/supabase-by-example/tree/main/react

**Comment by ChangJun2019** [comment]
@bflemi3 Hi, I had a similar issue. You can try upgrading @supabase/ssr to v0.4.1. In this version, someone has already fixed the related issues.

**Comment by bhimeshchauhan** [comment]
@bflemi3 were you able to resolve this?

**Reply by bflemi3** [reply]
I did not, but we ended up going in a different direction, so didn't get the time to work through the issue.
